tensorboard/webapp/metrics/views/card_renderer/scalar_card_component.ts

```diff
@@ -183,6 +192,48 @@ export class ScalarCardComponent<Downloader> {
     }
   }
 
+  getSelectedTimeTableData(): RunStepData[] {
+    if (this.selectedTime === null) {
+      return [];
+    }
+    const dataTableData = this.dataSeries
+      .map((datum) => {
+        const metadata = this.chartMetadataMap[datum.id];
+        const closestStartPoint = this.getClosestPoint(
+          datum.points,
+          this.selectedTime!.startStep
+        );
+        const closestEndPoint = null;
+        return {
+          metadata,
+          closestStartPoint,
+          closestEndPoint,
+        };
+      })
+      .filter(({metadata}) => {
+        return metadata && metadata.visible && !Boolean(metadata.aux);
+      });
+    return dataTableData;
```

Comment:
Now the input type looks like: `{metadata: {...}, closestStartPoint: {...}}`
I'm thinking about cleaning this before feeding to data table

Maybe a more fundamental question is, how/where do you plan to clean up things? In my opinion, the data table takes a list of columnHeaders and an object. It "prints" the value from the object if there are the keys matching the columnHeader.

which means, the input [data]=.. is something like
```
[{
run: ..
step: ...
}, {
run: ..
step: ...
}]
```
each item represents a row in the data table. What do you think?

Reply:
Can we add this ("get the required piece of information for each header") to the jsdoc of `tensorboard/webapp/widgets/data_table/data_table_component.ts`
Or added the comment above `interface RunStepData`?